frontend/app/src/components/settings/accounting/rule/AccountingRuleSetting.vue

```diff
@@ -204,45 +155,120 @@ function showDeleteConfirmation(item: AccountingRuleEntry) {
   );
 }
 
-function getType(eventType: string, eventSubtype: string) {
-  return get(
-    getEventTypeData({
-      eventSubtype,
-      eventType,
-    }),
-  );
+async function handleRuleAction(action: 'add-general' | 'add-event-specific' | 'edit-general' | 'edit-event-specific') {
+  const props = get(actionDialogProps);
+  if (!props)
+    return;
+
+  const { eventId, eventSpecificRule, generalRule } = props;
+
+  switch (action) {
+    case 'add-general': {
+      const { query } = get(route);
+      const { counterparty, eventSubtype, eventType } = query;
+      add({
+        ...getPlaceholderRule(),
+        counterparty: counterparty?.toString() ?? null,
+        eventSubtype: eventSubtype?.toString() ?? '',
+        eventType: eventType?.toString() ?? '',
```

Comment:
To avoid this repetition can we maybe have a zod schema that parses the query and returns the defaults if missing so that we don't have the repetition.

Also from what I can see the two cases are almost identical so can we avoid the repetition and use a common function maybe with and optional parameter